pkg/kubelet/network/plugins.go

```diff
@@ -47,6 +48,30 @@ type NetworkPlugin interface {
 
 	// TearDownPod is the method called before a pod's infra container will be deleted
 	TearDownPod(namespace string, name string, podInfraContainerID kubeletTypes.DockerID) error
+
+	// Status is the method called to obtain the ipv4 and ipv6 addresses of the container
+	Status(namespace string, name string, podInfraContainerID kubeletTypes.DockerID) (*PodNetworkStatus, error)
+}
+
+// InterfaceInfo is a structure to store network interface of a pod along with its associated IP addresses
+type InterfaceInfo struct {
+	NetworkInterface net.Interface `json:"networkInterface" description:"A network interface on the pod, e.g. eth1(index:1,mtu:1500,name:eth1..)"`
+	IPAddresses      []net.IP      `json:"ipAddresses" description:"List of IP Addresses associated with the given interface"`
+}
+
+// PodNetworkStatus stores the network status of a pod (currently just the network addresses and interfaces)
+// This struct represents version "v1"
+type PodNetworkStatus struct {
+	api.TypeMeta `json:",inline"`
+
+	// IPAddress is the primary ipv4/ipv6 address of the pod. Among other things it is the address that -
+	//   - kube expects to be reachable across the cluster
+	//   - service endpoints are constructed with
+	//   - will be reported in the PodStatus.PodIP field (will override the IP reported by docker)
+	IPAddress net.IP `json:"ipAddress" description:"Primary address of the pod"`
+
+	// NetworkInterfaces is a list of network interfaces that may be optionally provided
+	NetworkInterfaces []InterfaceInfo `json:"networkInterfaces,omitempty"`
```

Comment:
I hear the desire to carry more information, but here's my fear.  Things like this that get added but don't have anyone who uses them tend to rot and smell bad.  @jainvipin do we have an immediate use for these?  Maybe we can find a way to expose this info somewhere (it's going to be fun with "experimental" exposures)?  If we kicked this out of *this* PR and commited it in a PR that actually consumes it, would it be a problem?